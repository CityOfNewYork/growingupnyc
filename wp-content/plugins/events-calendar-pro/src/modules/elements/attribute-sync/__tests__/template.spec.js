import React from 'react';
import renderer from 'react-test-renderer';
import AttributeSync from '../template';

describe( 'AttributeSync', () => {
	test( 'should match snapshot', () => {
		const component = renderer.create(
			<AttributeSync
				clientId="1"
				initialize={ jest.fn() }
				cancel={ jest.fn() }
			/>
		);
		expect( component.toJSON() ).toMatchSnapshot();
	} );

	test( 'should call initialize and cancel functions', () => {
		const initialize = jest.fn();
		const cancel = jest.fn();
		const component = mount(
			<AttributeSync
				clientId="1"
				initialize={ initialize }
				cancel={ cancel }
			/>
		);
		expect( initialize ).toHaveBeenCalled();
		expect( initialize ).toHaveBeenCalledTimes( 1 );
		component.unmount();
		expect( cancel ).toHaveBeenCalled();
		expect( cancel ).toHaveBeenCalledTimes( 1 );
	} );
} );
