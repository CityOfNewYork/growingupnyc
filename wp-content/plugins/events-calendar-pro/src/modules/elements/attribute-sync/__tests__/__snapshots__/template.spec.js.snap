// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AttributeSync should match snapshot 1`] = `null`;
